src/notification/NotificationController.test.ts

```diff
@@ -0,0 +1,217 @@
+import { ControllerMessenger } from '../ControllerMessenger';
+import { GetSubjectMetadataState } from '../subject-metadata';
+import {
+  ControllerActions,
+  DismissNotification,
+  GetNotificationCount,
+  GetNotifications,
+  GetNotificationState,
+  GetUnreadNotificationCount,
+  MarkNotificationRead,
+  NotificationController,
+  NotificationMessenger,
+  NotificationStateChange,
+  NotificationType,
+  ShowNotification,
+} from './NotificationController';
+
+const name = 'NotificationController';
+
+/**
+ * Constructs a unrestricted controller messenger.
+ *
+ * @returns A unrestricted controller messenger.
+ */
+function getUnrestrictedMessenger() {
+  return new ControllerMessenger<
+    | GetNotificationState
+    | ShowNotification
+    | DismissNotification
+    | MarkNotificationRead
+    | GetNotificationCount
+    | GetUnreadNotificationCount
+    | GetNotifications
+    | GetSubjectMetadataState,
+    NotificationStateChange
+  >();
+}
+
+/**
+ * Constructs a restricted controller messenger.
+ *
+ * @param controllerMessenger - An optional unrestricted messenger
+ * @returns A restricted controller messenger.
+ */
+function getRestrictedMessenger(
+  controllerMessenger = getUnrestrictedMessenger(),
+) {
+  return controllerMessenger.getRestricted<
+    typeof name,
+    ControllerActions['type'] | GetSubjectMetadataState['type'],
+    never
+  >({
+    name,
+    allowedActions: [
+      'SubjectMetadataController:getState',
+      'NotificationController:show',
+    ],
+  }) as NotificationMessenger;
```

Comment:
Why are we typecasting this?

Reply:
I was not able to get it to stop complaining about the types without doing this. There was a similar problem when testing the PermissionsController iirc.

Reply:
This is probably because the restricted messenger used by the tests is incorrect. It should have no allowed actions, as declared in the notification messenger. You don't need to list a controllers own actions as allowed, they are automatically allowed by virtue of being under the same namespace.

Reply:
Fixed!